### ISSUE TITLE
Please how to add latest net of katago

### ISSUE THREAD

**Issue by hydrogenpi** [lightvector/KataGo#136]
Please how to add latest net of katago

to sabaki
and to Lizzie

**Comment by lightvector** [comment]
Sorry, I don't understand. If you have an older version of KataGo working with Sabaki or Lizzie, just download a newer release (https://github.com/lightvector/KataGo/releases) and replace the executable, and download a newer neural net model and replace your older `.txt.gz` neural net model file. Are you asking about something besides this?

**Comment by lightvector** [comment]
And once it's working, if you want best performance and newer parameters, you can also replace your old gtp.cfg, and re-tune threads, as described here: https://github.com/lightvector/KataGo#tuning-for-performance
(although this should not be necessary for merely getting it working)

**Comment by nightsfan** [comment]
Firstly, thanks for making Katago!  It's brilliant :)

With Lizzie, I have the same problem: I can easily update the LZ neural networks on Lizzie (just by replacing the existing LZ weights and keeping the filename the same), but Katago is much harder to get working on Lizzie, if you want the new weights to work.  

The steps I tried (of which the first 3 work fine) were:

1. Download/extract Lizzie and test run both LZ and (pre-installed) Katago networks (both work)
2. Download/extract newest Katago and run tuning via Windows command prompt (this took about 20 minutes to complete), following guide on the Katago website (this worked ok)
3. Copied the executable (katago.exe) from Katago folder and then replaced the executable in Lizzie's Katago subfolder with it (I left all other files in the subfolder unchanged) {I did a test run at this stage: with the new Katago executable, it still runs OK in Lizzie!)
4. Copied the new 30-block Katago network ('default_model.bin.gz') to the Lizzie folder, deleted the existing 'katanetwork.gz', and renamed default_model.bin.gz to 'katanetwork.gz'

After step 4, when I launch Lizzie and select Katago, it just says 'Engine is loading' (indefinitely).

I wonder if a fix might require manually editing certain config files or filenames (should I rename Lizzie's current katagonetwork.gz to 'default_model.bin.gz' or 'katanetwork.bin.gz' (?) and then manually edit the Lizzie config file to look for this filename instead? Or is Katago doing the very long self-tuning again, without saying so? -- I guess not as Lizzie has done initial Katago tuning, and the newest Katago was also self-tuned before copying stuff over to Lizzie).

I'm not at all familiar with coding or computing (e.g. using command prompts or editing files/paths), so I'm sorry if this is missing something very obvious! 

(System: I'm using Windows 10, with a recent i7 cpu and a basic 10th gen Geforce card.)

**Comment by lightvector** [comment]
> I wonder if a fix might require manually editing certain config files or filenames (should I rename Lizzie's current katagonetwork.gz to 'default_model.bin.gz' or 'katanetwork.bin.gz' (?) and then manually edit the Lizzie config file to look for this filename instead? Or is Katago doing the very long self-tuning again, without saying so? -- I guess not as Lizzie has done initial Katago tuning, and the newest Katago was also self-tuned before copying stuff over to Lizzie).

You cannot remove or shorten ".bin.gz" part to just ".gz", currently KataGo uses that to determine how to load the file (since there is more than one model format). So yes, you need to edit Lizzie's config. Additionally, yes, it will *also* need to do the long self-tuning, it will usually need to do this each time you change to an entirely new model size, once for each size.

The way to debug something like this is *always* to run the program directly yourself in the command line, as recommended in the KataGo front page: https://github.com/lightvector/KataGo#how-to-use. It is still the case, sadly, that all GUIs are very very bad at reporting mistakes in configuring things and if you have anything wrong, they won't tell you what is wrong, whereas if you run it directly on command line, you will actually see an error message telling you what is wrong (or you will see it tuning and therefore nothing is wrong, it just takes time, etc).

It sounds like the great majority all the steps you are doing related to Lizzie being prepackaged with an old version of KataGo. Hopefully at some point in the future when Lizzie updates it will come with a newer version and then most of this becomes unnecessary.

**Comment by GopherAl** [comment]
@nightsfan You could try LizGoban as an alternative to Lizzie. (Lizzie hasn't been updated since last September.) LizGoban is (at least partly) based on Lizzie. It has up-to-date KataGo nets built into it similarly to Lizzie.

**Comment by lightvector** [comment]
Did you eventually get this working?

**Comment by lightvector** [comment]
KataGo as of v1.4.0 (and v1.4.1) will tolerate files named to just ".gz", which should make at least one of the above steps easier. Yay.

It's been more than a month now since the last post, so I assume you've gotten it working or given up. If you gave up, you might consider trying again, or trying some of the suggestions and then asking questions again once you run into new errors. It's a little hard to keep helping otherwise. :)

Anyways, closing this issue, but please comment back or open a new issue if you try again and run into problems. Or, try the Leela Zero discord chat (which hosts KataGo and other bots) if you want to find someone to help you with it interactively in real time - which can be very nice and much easier.
https://discord.gg/MqQfA8q

**Comment by nightsfan** [comment]
Thanks so much for this!  I'm sorry, I had not quite given up - sorry for
the late reply!  I was planning to have a go again this week.  Recently, I
had switched to just using another program to run Katago (Lizgoban), as
Lizzie had failed each time that I tried modifying the config files
before.  I will try again now; thanks to your update, it should be even
easier :)  I'll also try the LZ discord chat if I still have problems
getting it going!  (Thanks for letting me know about this kind of thing!  I
had no idea about such things.)  It will be great to have Katago running on
Lizzie, as it's still the most user-friendly GUI (Lizgoban is interesting
but it's visually a bit confusing with all the lines and rainbow colours).

Again, my apologies for the delay, and thanks for your kind help! ^_^

**Comment by nightsfan** [comment]
OK, I managed to resolve the issue, thanks to your help! With the 1.4 update (which tolerates filenames just ending in .gz), the copy and paste method works find, without any need to manually edit config files etc.  :)

So I am now running the latest 20-block network on 1.4 through Lizzie.  Thanks again for the help and the 1.4 update!  Much easier now :D
